File: server/src/modules/users/domain/entities/user.entity.ts
export class UserEntity {
  constructor(
    public readonly id: string,
    public readonly phone: string | null,
    public readonly identityNumber: string | null,
    public readonly fullName: string,
    public readonly avatarUrl: string | null,
    public readonly role: RoleType[],
    public readonly createdAt: string,
    public readonly updatedAt: string,
    public readonly acceptedTerms: boolean,
    public readonly dateOfBirth: string | null = null,
  ) {}
}

File: client/packages/domain/lib/src/auth/entities/user_entity.dart
abstract class UserEntity {
  UserEntity({
    required this.id,
    this.email,
    required this.role,
    this.phone,
    this.fullName,
    this.avatarUrl,
    this.createdAt,
    this.updatedAt,
    this.acceptedTerms,
  });

  final String id;
  final String? email;
  final dynamic role;
  final String? phone;
  final String? fullName;
  final String? avatarUrl;
  final String? createdAt;
  final String? updatedAt;
  final bool? acceptedTerms;
}
